lib/demos/todo.dart

```diff
@@ -197,3 +189,84 @@ class TodoAppState extends State<TodoApp> {
     scheduleUpdate();
   }
 }
+
+
+typedef void ChangeListener();
+
+/// A simple observable model object.
+///
+/// Extend it to get concrete observable objects.
+class Model {
+  List<ChangeListener> _listeners;
+
+  void addListener(ChangeListener listener) {
+    _listeners ??= <ChangeListener>[];
+    _listeners.add(listener);
+  }
+
+  void removeListener(ChangeListener listener) {
+    if (_listeners == null) return;
+    _listeners.removeWhere((l) => l == listener);
+  }
+
+  void objectDidChange() {
+    if (_listeners == null) return;
+    for (var listener in _listeners) {
+      listener();
+    }
+  }
+}
+
+abstract class KeyModel extends Model {
+  KeyModel(this.key);
+
+  final num key;
+}
+
+class Todo extends KeyModel {
+  Todo(num key, this._title, this._completed) : super(key);
+
+  String _title;
+  String get title => _title;
+  set title(String newTitle) {
+    if (newTitle != _title) {
+      _title = newTitle;
+      objectDidChange();
+    }
+  }
+
+  bool _completed;
+  bool get completed => _completed;
+  set completed(bool newCompleted) {
+    if (newCompleted != _completed) {
+      _completed = newCompleted;
+      objectDidChange();
+    }
+  }
+}
+
+class TodoFactory {
+  int _uid = 0;
+
+  int nextUid() => ++_uid;
+
+  Todo create(String title, bool isCompleted) {
+    return new Todo(this.nextUid(), title, isCompleted);
+  }
+}
+
+class Store<T extends KeyModel> {
+  List<T> list = <T>[];
+
+  void add(T record) {
+    list.add(record);
+  }
+
+  void remove(T record) {
+    list.remove(record);
+  }
+
+  void removeBy(bool callback(T t)) {
```

Comment:
This is a straight port from the Angular/TypeScript version. Lots of undarty things here. Not worried too much though. This is only a sample. The "Store" is kinda pointless too. It implies there's some FRP going on, but in fact it's just a list of todos. Could simply be a `List`. Guessing they wrote it in TypeScript because there are no standard human-friendly collections, only Array.